gumroad/src/gumroad-block/block.js

```diff
@@ -0,0 +1,256 @@
+  render() {
+    const {
+      attributes: {
+        id,
+        text,
+        type,
+        url,
+        button,
+        classes,
+        wanted
+      },
+      setAttributes
+    } = this.props;
+
+    return[
+      <GumControls { ...{setAttributes, ...this.props } } />,
+      <div id="gumroad-block" className="gumroad-block">
+        <RichText
+          tagName="div"
+          placeholder={ __( 'I want this!' ) }
+          keepPlaceholderOnFocus
+          className="gumroad-block-link"
+          value={ text }
+          onChange={ ( value ) => setAttributes( { text: value } ) }
+        />
+      </div>
+    ];
+  }
+};
+
+/**
+ * Register: aa Gutenberg Block.
+ *
+ * Registers a new block provided a unique name and an object defining its
+ * behavior. Once registered, the block is made editor as an option to any
+ * editor interface where blocks are implemented.
+ *
+ * @link https://wordpress.org/gutenberg/handbook/block-api/
+ * @param  {string}   name     Block name.
+ * @param  {Object}   settings Block settings.
+ * @return {?WPBlock}          The block, if it has been successfully
+ *                             registered; otherwise `undefined`.
+ */
+registerBlockType( 'gumroad/gumroad-block', {
+	// Block name. Block names must be string that contains a namespace prefix. Example: my-plugin/my-custom-block.
+	title: __( 'Gumroad Block' ), // Block title.
+	icon: 'shield', // Block icon from Dashicons → https://developer.wordpress.org/resource/dashicons/.
+	category: 'common', // Block category — Group blocks together based on common traits E.g. common, formatting, layout widgets, embed.
+	keywords: [
+		__( 'gumroad-block' ),
+	],
+  attributes : {
+    id: {
+      type: 'string',
+      default: ''
+    },
+    type: {
+      type: 'string',
+      default: 'none'
+    },
+    classes: {
+      type: 'string',
+      default: ''
+    },
+    text: {
+      type: 'string',
+      default: 'I want this!'
+    },
+    wanted: {
+      type: 'string',
+      default: ''
+    },
+    button: {
+      type: 'string',
+      default: 'false'
+    },
+    url: {
+      type: 'string',
+      default: ''
+    }
+  },
+
+	/**
+	 * The edit function describes the structure of your block in the context of the editor.
+	 * This represents what the editor will render when the block is used.
+	 *
+	 * The "edit" property must be a valid function.
+	 *
+	 * @link https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/
+	 */
+	edit: EditBlockContent,
+
+	/**
+	 * The save function defines the way in which the different attributes should be combined
+	 * into the final markup, which is then serialized by Gutenberg into post_content.
+	 *
+	 * The "save" property must be specified and must be a valid function.
+	 *
+	 * @link https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/
+	 */
+	save: function( props ) {
+    const { attributes: { id, type, classes, text, wanted, button, url }} = props;
```

Comment:
spacing here looks janky

Reply:
Looks like tab display is different in my editor. I'll clean it up.